_posts/2021-10-12-devfile-v2-and-the-devworkspace-operator-p1.md

```diff
@@ -0,0 +1,149 @@
+
+### A new way to specify the editor and its plugins
+As mentioned above, version 2 of the Devfile, doesn’t include cheEditor and chePlugins component types anymore. 
+
+The recommended way to specify the editor of a workspace is to include the file `.che/che-editor.yaml` at the root of the workspace git repository:
+```yaml
+id: eclipse/che-theia                 # mandatory
+registryUrl: https://my-registry.com  # optional
+override:                             # optional
+  containers:
+    - name: theia-ide
+      memoryLimit: 1280Mi
+```
+
+The recommended way to specify a che-theia plugin in a workspace is to include the file `.vscode/extensions.json` at the root of the workspace git repository:
+```json
+{
+    "recommendations": [
+      "ms-python.python"
+    ]
+}
+```
+ 
+It is also possible to define a Che editor and its plugins inline in Devfile attributes, but that’s recommended only when it is not possible to colocate the devile with the source code. More informations about IDEs plugins with v2 of the Devfile can be found [in this blog post](https://che.eclipseprojects.io/2021/06/23/@florent.benoit-devfile-v2-and-ide-plug-ins.html) and at [https://github.com/eclipse/che/issues/18669](https://github.com/eclipse/che/issues/18669).
+
+### The Devfile should live at the root of the git repo, not in a registry
+The recommended place to publish the Devfile is within the project source code. Along with the files that we have just seen above to specify che-theia plugins and the editor:
+
+```
+  |--- devfile.yaml
+  |___ .che
+         |___ che-editor.yaml
+         |___ che-theia-plugins.yaml
+  |___ .vscode
+         |___ extensions.json
+```
+
+Having the Devfile at the root of a repository makes it possible to use a simple factory link to the repository to start the workspace. For example [workspaces.openshift.com/#https://github.com/eclipse/che-docs](https://workspaces.openshift.com/#https://github.com/eclipse/che-docs) is a factory link: when a user click on it, a workspace defined by the Devfile in the repo [github.com/eclipse/che-docs](https://github.com/eclipse/che-docs) will be started on the public Che instance hosted by Red Hat at workspaces.openshift.com. 
+
+Recommending to colocate and evolve a v2 Devfile with the source code has two consequences:
+- the `project` section of a Devfile can be omitted: it’s implicitly set to the git repo where the Devfile lives
+- Che samples will include a Devfile at their root (those Devfiles used to be published in the Devfile registry)  
+
+### Only one running workspace per user
+A user cannot have more than one running workspace at a time. This limitation is related to the persistent volume strategy (“common”) that is used by Che. The same Volume is mounted by every workspace of the same user. This is implemented using Pods `volumeMounts.subPath` property and guarantees that the number of Volumes mounted by Che matches the number of users.
```

Comment:
> The same Volume is mounted by every workspace of the same user

Volume -> PersistentVolume not to initiate confusion with devfile volume

> This is implemented using Pods `volumeMounts.subPath` property and guarantees that the number of Volumes mounted by Che matches the number of users.

This mixes two stuff:
- volumeMounts.subPath: the way we isolate files of different workspace in one PV
- show a motivation why we stick to one PV per user.